### PR TITLE
Add Keywords AI (YC W24)

### PR DESCRIPTION
**Pull request by YunhaoLiu1228**
## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by sonichi** [comment]
Thanks. Could you please fix code formatting error? https://github.com/microsoft/autogen/actions/runs/9335787253/job/25706192247?pr=2842
Using pre-commit might help.

**Comment by marklysze** [comment]
Hi @YunhaoLiu1228, thanks for putting this together, I hadn't heard of Keywords AI and it looks like a good platform to be able to interact with many different providers.

I signed up and tested the notebook and it worked as expected.

Is it possible to update the notebook so it doesn't show an error output? I believe it's because Enter was pressed for the last user prompt, I'd suggest you type: exit

File: website/docs/topics/non-openai-models/cloud-keywordsai.ipynb [hunk 1]
```diff
@@ -0,0 +1,188 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Keywords AI\n",
+    "[Keywords AI](https://keywordsai.co/) is a unified developer platform where you can call 150+ LLM using the OpenAI format with one API key and get insights into your AI products. With 2 lines of code, you can build better AI products with complete observability.\n",
```
Comment (marklysze) [inline website/docs/topics/non-openai-models/cloud-keywordsai.ipynb:8]:
```suggestion
    "[Keywords AI](https://keywordsai.co/) is a unified developer platform where you can call 150+ LLMs, using the OpenAI API format, with one API key and get insights into your AI products. With 2 lines of code, you can build better AI products with complete observability.\n",
```

File: website/docs/topics/non-openai-models/cloud-keywordsai.ipynb [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      "\u001b[0;31mBadRequestError\u001b[0m: Error code: 400 - {'detail': 'Invalid input params: \\nmessages: Value error, Empty string not allowed for content, received: \\n'}"
```
Comment (marklysze) [inline website/docs/topics/non-openai-models/cloud-keywordsai.ipynb:141]: If you can re-run and exit to terminate rather than use Enter.

**Comment by r4881t** [comment]
@YunhaoLiu1228 - Will this work only for OpenAI models?

I tried to use Gemini model with keywordsAI using the following llm config

```python
model_gemini = "gemini-pro"
llm_config_keywordsai = {
  "config_list": [{"model": model_gemini, "api_key": os.environ["KEYWORDSAI_API_KEY"]}],
  "stream": False,
  "temperature": 0,
  "cache_seed": 20,
}
```

And I encountered error in `ConversableAgent` as below

```
  File "/Users/pranavprakash/workspace/friday/venv/lib/python3.11/site-packages/autogen/agentchat/conversable_agent.py", line 1340, in generate_oai_reply
    extracted_response = self._generate_oai_reply_from_client(
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pranavprakash/workspace/friday/venv/lib/python3.11/site-packages/autogen/agentchat/conversable_agent.py", line 1359, in _generate_oai_reply_from_client
    response = llm_client.create(
               ^^^^^^^^^^^^^^^^^^
  File "/Users/pranavprakash/workspace/friday/venv/lib/python3.11/site-packages/autogen/oai/client.py", line 639, in create
    response = client.create(params)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pranavprakash/workspace/friday/venv/lib/python3.11/site-packages/autogen/oai/client.py", line 285, in create
    response = completions.create(**params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pranavprakash/workspace/friday/venv/lib/python3.11/site-packages/openai/_utils/_utils.py", line 299, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pranavprakash/workspace/friday/venv/lib/python3.11/site-packages/openai/resources/chat/completions.py", line 598, in create
    return self._post(
           ^^^^^^^^^^^
  File "/Users/pranavprakash/workspace/friday/venv/lib/python3.11/site-packages/openai/_base_client.py", line 1063, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pranavprakash/workspace/friday/venv/lib/python3.11/site-packages/openai/_base_client.py", line 842, in request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/pranavprakash/workspace/friday/venv/lib/python3.11/site-packages/openai/_base_client.py", line 885, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.AuthenticationError: Error code: 401 - {'error': {'message': 'Incorrect API key provided: zarCTmvb*****************************8Erm. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}
```

I believe this is due to the check of API Key (both in Autogen & in OpenAI packages), so they expect this in a specific format

**Comment by Deykeepcomn4me** [comment]
These are hackers hacking me

**Comment by YunhaoLiu1228** [comment]
Sorry for the inconvenience. We’ll fix it this week!

Best,
Yunhao

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by gagb** [comment]
@YunhaoLiu1228 thank you! Can you please update the PR if you still would like this to be included?

**Pull request closed without merging**